libpod/image/pull.go

```diff
@@ -356,11 +420,36 @@ func checkRemoteImageForLabel(ctx context.Context, label string, imageInfo pullR
 	if err != nil {
 		return err
 	}
+	return isLabelInInspect(label, remoteInspect.Labels, imageInfo.image)
+}
+
+// checkRemoteImageIsUpToDate checks if the remote image has a specific label. If not, an error is returned.
```

Comment:
It’s a bit surprising to combine the label presence check with an up-to-date check this way, although, sure, the `NewImage`+`Inspect` calls are costly. At least mention it in the name?